pkgs/development/compilers/zig/generic.nix

```diff
@@ -92,16 +98,29 @@ stdenv.mkDerivation (finalAttrs: {
   # Zig's build looks at /usr/bin/env to find dynamic linking info. This doesn't
   # work in Nix's sandbox. Use env from our coreutils instead.
   postPatch =
-    if lib.versionAtLeast finalAttrs.version "0.12" then
-      ''
-        substituteInPlace lib/std/zig/system.zig \
-          --replace-fail "/usr/bin/env" "${coreutils}/bin/env"
-      ''
-    else
-      ''
-        substituteInPlace lib/std/zig/system/NativeTargetInfo.zig \
-          --replace-fail "/usr/bin/env" "${coreutils}/bin/env"
-      '';
+    (
+      if lib.versionAtLeast finalAttrs.version "0.12" then
+        ''
+          substituteInPlace lib/std/zig/system.zig \
+            --replace-fail "/usr/bin/env" "${coreutils}/bin/env"
```

Comment:
```suggestion
            --replace-fail "/usr/bin/env" "${lib.getExe' coreutils "env"}"
```

Reply:
Addressed in 7d07c588d583491cbc8a3a112bd060db20bf9001